client/src/main/java/io/split/client/impressions/ImpressionObserver.java

```diff
@@ -0,0 +1,38 @@
+package io.split.client.impressions;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import io.split.client.dtos.KeyImpression;
+
+/*
+According to guava's docs (https://guava.dev/releases/18.0/api/docs/com/google/common/annotations/Beta.html),
+the @Beta decorator only means that the api is not frozen, and has nothing to do with behaviour stability, but
+rather to a non-frozen API which may introduce breaking changes at any time in future versions.
+Since the library is shaded and should not be exposed to users of the SDK, it's safe to use it here.
+ */
+
+@SuppressWarnings("UnstableApiUsage")
+public class ImpressionObserver {
+
+    private final Cache<Long, Long> _cache;
+
+    public ImpressionObserver(long size) {
+        _cache = CacheBuilder.newBuilder()
```

Comment:
do we need to set the concurrency level?

Reply:
how about the expiration settings?

Reply:
Regarding cuncurrency level: the cache is only used from the impressions manager which is executed within a singleThreadedExecutor, so we should not have race conditions here, nor benefit from partitioning to minimize contention. I can however set the default value (4) explicitly, in case it changes sometime, or we decide to go multithreaded here

Regarding expiration settings: We discussed this thoroughly with nico, and in order to minimize memory usage, we decided not to attach an extra TTL to each cache item, since the difference between sending `null` or sending an old timestamp doesn't really make a difference. We let the events server decide whether the timestamp is recent enough to filter the impression from the exp pipeline or not